src/lib.rs

```diff
@@ -238,18 +245,33 @@ impl<'a> Ammonia<'a> {
     }
 
     fn fix_child(&self, child: &mut Handle, link_rel: &Option<StrTendril>, url_base: &Option<Url>) {
-        if let (&NodeData::Element{ref name, ref attrs, ..}, &Some(ref link_rel)) = (&child.data, link_rel) {
-            if &*name.local == "a" {
-                attrs.borrow_mut().push(Attribute{
-                    name: QualName::new(None, ns!(), local_name!("rel")),
-                    value: link_rel.clone(),
-                })
+        if let &NodeData::Element{ref name, ref attrs, ..} = &child.data {
+            if let &Some(ref link_rel) = link_rel {
+                if &*name.local == "a" {
+                    attrs.borrow_mut().push(Attribute{
+                        name: QualName::new(None, ns!(), local_name!("rel")),
+                        value: link_rel.clone(),
+                    })
+                }
+                if let &Some(ref base) = url_base {
+                    for attr in &mut *attrs.borrow_mut() {
+                        if is_url_attr(&*name.local, &*attr.name.local) {
+                            let url = base.join(&*attr.value).expect("invalid URLs should be stripped earlier");
+                            attr.value = format_tendril!("{}", url);
+                        }
+                    }
+                }
             }
-            if let &Some(ref base) = url_base {
+            if self.allowed_classes.contains_key(&*name.local) {
                 for attr in &mut *attrs.borrow_mut() {
-                    if is_url_attr(&*name.local, &*attr.name.local) {
-                        let url = base.join(&*attr.value).expect("invalid URLs should be stripped earlier");
-                        attr.value = format_tendril!("{}", url);
+                    if &attr.name.local == "class" {
+                        let mut classes = vec![];
+                        for class in attr.value.split(' ') {
+                            if self.allowed_classes[&*name.local].contains(class) {
```

Comment:
Well we check that the element is in the map above. `if self.allowed_classes.contains_key(&*name.local)`

Maybe I can do something better with .get() and pattern matching, I'll try that :)